ST_grid.c

```diff
@@ -706,2357 +390,1206 @@ static void _init_grid_files(void)
-		memcpy(SW_Site.lyr[j], spinup_SW_Site[cell].lyr[j],
-				sizeof(SW_LAYER_INFO));
-	}
-
-	if (UseSoils)
-		load_sxw_memory(spinup_SXW_ptrs[cell].roots_max,
-				spinup_SXW_ptrs[cell].rootsXphen,
-				spinup_SXW_ptrs[cell].roots_active,
-				spinup_SXW_ptrs[cell].roots_active_rel,
-				spinup_SXW_ptrs[cell].roots_active_sum,
-				spinup_SXW_ptrs[cell].phen,
-				spinup_SXW_ptrs[cell].prod_bmass,
-				spinup_SXW_ptrs[cell].prod_pctlive);
+    ChDir(grid_directories[GRID_DIRECTORY_STEPWAT_INPUTS]);
+    parm_SetName(grid_files[GRID_FILE_MAXRGROUPSPECIES], F_MaxRGroupSpecies);
+    maxrgroupspecies_init();
+    ChDir("..");
 }
 
-/***********************************************************/
-static void _save_cell(int row, int col, int year, Bool useAccumulators)
+/* Read the non-gridded mode files.in file. */
+static void _read_files(void)
 {
-	// saves the specified cell into the grid variables
+    ChDir(grid_directories[GRID_DIRECTORY_STEPWAT_INPUTS]);
+    files_init();
+    ChDir("..");
+}
 
-	int cell = col + ((row - 1) * grid_Cols) - 1; // converts the row/col into an array index
-	int j, k;
-	GrpIndex c;
-	SppIndex s;
-	//fprintf(stderr, "saving cell: %d\n", cell);
+/* Reads the grid setup file and allocates gridCells.*/
+static void _read_grid_setup(void)
+{
+    FILE *f;
+    char buf[1024], initializationType[1024];
+    int i, j;
 
-	if (useAccumulators)
-		stat_Save_Accumulators(cell, year);
+    f = OpenFile(grid_files[GRID_FILE_SETUP], "r");
 
-	ForEachSpecies(s)
-	{
-		if (!Species[s]->use_me)
-			continue;
+    GetALine(f, buf);
+    i = sscanf(buf, "%d %d", &grid_Rows, &grid_Cols);
+    if (i != 2)
+        LogError(logfp, LOGFATAL,
+                 "Invalid grid setup file (rows/cols line wrong)");
 
-		Mem_Free(grid_Species[s][cell].kills);
-		Mem_Free(grid_Species[s][cell].seedprod);
-		_free_head(grid_Species[s][cell].IndvHead);
+    grid_Cells = grid_Cols * grid_Rows;
 
-		grid_Species[s][cell] = *Species[s];
+    if (grid_Cells > MAX_CELLS)
+        LogError(logfp, LOGFATAL,
+                 "Number of cells in grid exceeds MAX_CELLS defined in ST_defines.h");
 
-		grid_Species[s][cell].kills = Mem_Calloc(Species[s]->max_age,
-				sizeof(IntUS), "_save_cell(grid_Species[cell][s].kills)");
-		grid_Species[s][cell].seedprod = Mem_Calloc(Species[s]->viable_yrs,
-				sizeof(RealF), "_save_cell(grid_Species[cell][s].seedprod)");
+    /* Allocate the 2d array of cells now that we know how many we need */
+    _allocate_gridCells(grid_Rows, grid_Cols);
 
-		memcpy(grid_Species[s][cell].kills, Species[s]->kills,
-				Species[s]->max_age * sizeof(IntUS));
-		memcpy(grid_Species[s][cell].seedprod, Species[s]->seedprod,
-				Species[s]->viable_yrs * sizeof(RealF));
-		grid_Species[s][cell].IndvHead = _copy_head(Species[s]->IndvHead);
-	}
+    GetALine(f, buf);
+    i = sscanf(buf, "%u", &UseDisturbances);
+    if (i != 1)
+        LogError(logfp, LOGFATAL,
+                 "Invalid grid setup file (distur
```

Comment:
MAXDP and maxCells are not in the code anymore - we need to retain all of the old seed dispersal implementation so we can easily move forward in the next development branch. I am leaning towards retaining this whole function for the time being, so we don't lose any of the existing functionality, even though it is poorly implemented.

Reply:
I added these variables back to the initDispersalParameters function and added the lines associated with them.
Resolved by 0291723.